packages/core/src/index.ts

```diff
@@ -23,6 +24,7 @@ try {
     checkRowLevelSecurity(EnvSet.queryClient),
     checkAlterationState(await EnvSet.pool),
   ]);
+  await SystemContext.shared.loadStorageProviderConfig(await EnvSet.pool);
```

Comment:
nit: since it's shared i think we can even directly define a static shared pool inside the class, like:

```ts
class SystemContext {
  static pool = EnvSet.pool;
}
```

Reply:
this pool has global access, I prefer not to store it in case it is being used unexpectly.